README.md

```diff
@@ -421,21 +421,56 @@ The `eval` function should be avoided.
 
 [⬆ back to TOC](#table-of-contents)
 
-##undefined
-Checking for `undefined` values should be done using the strict equality operator.
+## undefined
+
+Checking for `undefined` values should be done using the strict equality operator:
+
+> Explanation:
+> * In modern browsers (`IE9+`, `Opera 12.16+`, `Firefox 4+`) [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined)
+>   is immutable (a non-configurable, non-writable property of the global object).
+> * It prevents using variable before declaration.
 
 **Good:**
+
 ```js
-x === undefined;
+x === undefined
 ```
 
 **Bad:**
+
 ```js
 typeof x === 'undefined'
-
 x === void 0
 ```
 
+**Exceptions:**
+
+  * `typeof` should be used if you need to support old browsers (like `IE8`) where `window.undefined` property is mutable.
+  * `typeof` should be used for detect undeclared global variables in **different** environments:
+
+  ```js
+  // It works in browser environment only.
+  if (window.feature !== undefined) {
```

Comment:
MDN [recommends](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#Typeof_operator_and_undefined) (?) use checking property of global object:

> ...but the global scope is bound to the global object, so checking the existence of a variable in the global context can be done by checking the existence of a property on the global object (using the in operator, for instance).

Personally I prefer use
```js
if (typeof Int32Array !== 'function') {}
```
instead
```js
if (typeof window.Int32Array !== 'function') {}
```
And I don't see here difference with `undefined` checking. Should we use typeof with global variables?